### ISSUE TITLE
Can we extend the humanize function to allow for booleans?

### ISSUE THREAD

**Issue by dlobo** [phoenixframework/phoenix_html#302]
Currently, ecto/phoenix has no convention for boolean fields in the database.

If we permit the is_* convention in the DB, and make the following tweak to humanize:

is_reserved ==> Reserved?

will make things a lot easier, rather than folks having to go in and manually change the templates

Happy to try and submit a patch to extend this. This will be similar to the _id convention

**Comment by josevalim** [comment]
This is assuming though that we use the "is_foo" convention in Elixir but that's not quite true. :) "is_foo" is recommended only for guards and structs/schemas fields are not guards.

**Comment by kelvinst** [comment]
I agree it would be a conflicting convention, but I think having a naming convention for booleans could be beneficial. Thinking a bit more I got to this option: changing the generator to translate `mix phx.gen.schema Schema foo?` to generate a boolean column "is_foo" on the table and a `field :foo?, :boolean, source: :is_foo` on the schema. Maybe even changing Ecto's `field` macro to translate a `:foo?` field to a `:is_foo` column automatically (as I think most db's don't accept `?` on column names anyway). WDYT? Am I going too far?

**Comment by josevalim** [comment]
If it is `foo?`, then this patch is most likely not required. And if it is "field :foo?", do we care what it actually is in the database? 🤔 what is the general DB convention for booleans?

**Comment by kelvinst** [comment]
Yeah, not the suggested one, my suggestion would be just to change the generator, as `foo?` is not a valid column name for most dbs, then in the migration it should be another name. 

In Rails, I used to use `is_foo` for boolean fields, not sure if it was a convention I got randomly or if it was actually enforced by the community.

Generally, I don't think there is any naming convention for boolean columns, so maybe we should not introduce one, as some conventions might create discussions (like this one), and the hassle of actually changing the templates or the field name is really not the big.
